src/types/coursDetails.type.ts

```diff
@@ -0,0 +1,9 @@
+export type coursDetails = {
+    id: number,
+    title: string,
+    level: string,
+    description: string,
+    rating: number,
+    nbrOfRaters: number,
+    trainer: string
```

Comment:
IMO, it should be of type Trainer/Instructor
```suggestion
    trainer: Trainer
```